elftools/construct/debug.py

```diff
@@ -38,9 +42,9 @@ class Probe(Construct):
     ]
     counter = 0
 
-    def __init__(self, name = None, show_stream = True,
-                 show_context = True, show_stack = True,
-                 stream_lookahead = 100):
+    def __init__(self, name: str | None = None, show_stream: bool = True,
```

Comment:
OOC: is it actually required to annotate `__init__` as returning None? Will things fail if you don't?

Reply:
Since [mypy 0.640 "New Feature: Allow Omitting Return Type for __init__"](https://mypy-lang.blogspot.com/2018/10/mypy-0640-released.html) (~2018-10) it can be omitted – as long as you have *at least one argument annotated* – a completely un-annotated function/method is considered _dynamically types_ and is **not** checked. I have not checked others type-checkers like `pytype` or `pyright` how they handle it.

For consistency I prefer to always type the return type, even for `__init__(…) -> None`